ONSdigital/eq-survey-runner
# Eq 322 harden security

### Changes

Hardening the security around the save and resume functionality, specifically:
1. Use a random IV
2. Generating a user id using a key derivation function
3. Generate a CEK passed on a salt and the generated user id
4. Enabling Flask database sessions to store the user id (the cookie will now only include the session id)
### How to test
1. Enable server side storage using the following env variables
   `export EQ_SERVER_SIDE_STORAGE=True`
   `export EQ_SERVER_SIDE_STORAGE_ENCRYPTION=True`
2. Clear all eq related cookies
3. Run the server and login
4. Check that in the eq cookie you can only see a session id
5. Check that all functionality of a survey still works
6. Check that you can partially complete a survey, get a new JWT and retrieve your old data.
### Who can test

Anyone apart from @warrenbailey
